src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -381,25 +381,23 @@ public abstract class SendReceiveTest : ClientWebSocketTestBase
         {
             var options = new LoopbackServer.Options { WebSocketEndpoint = true };
 
-            Func<ClientWebSocket, Socket, Uri, Task> connectToServerThatAbortsConnection = async (clientSocket, server, url) =>
+            Func<ClientWebSocket, LoopbackServer, Uri, Task> connectToServerThatAbortsConnection = async (clientSocket, server, url) =>
             {
                 AutoResetEvent pendingReceiveAsyncPosted =  new AutoResetEvent(false);
 
                 // Start listening for incoming connections on the server side.
-                Task<List<string>> acceptTask = LoopbackServer.AcceptSocketAsync(server, async (socket, stream, reader, writer) =>
+                Task acceptTask = server.AcceptConnectionAsync(async connection =>
                 {
                     // Complete the WebSocket upgrade. After this is done, the client-side ConnectAsync should complete.
-                    Assert.True(await LoopbackServer.WebSocketHandshakeAsync(socket, reader, writer));
+                    Assert.True(await LoopbackHelper.WebSocketHandshakeAsync(connection));
 
                     // Wait for client-side ConnectAsync to complete and for a pending ReceiveAsync to be posted.
                     pendingReceiveAsyncPosted.WaitOne(TimeOutMilliseconds);
 
                     // Close the underlying connection prematurely (without sending a WebSocket Close frame).
-                    socket.Shutdown(SocketShutdown.Both);
-                    socket.Close();
-
-                    return null;
-                }, options);
+                    connection.Socket.Shutdown(SocketShutdown.Both);
+                    connection.Socket.Close();
```

Comment:
This will be done by the disposal of the Connection in AcceptConnectionAsync, right?  Doesn't hurt to be explicit about it of course if it's critical to the test.

Reply:
Not sure if it's critical, but since this test is a bit unusual I just left it as is.